lib/properties.js

```diff
@@ -111,29 +151,27 @@ class Properties extends EventEmitter {
     this._building = setTimeout(() => {
       // Merge layers into their own object. This can be consumed as an input by
       // template renderers.
-      const persistent = this.layers.reduce((p, layer) => {
+      const persistent = this.layers.reduce((properties, layer) => {
         if (!layer.namespace) {
-          return p.mergeDeep(Immutable.fromJS(layer.properties));
+          return merge(properties, layer.properties);
         }
-        const layerProps = Immutable.Map(layer.properties);
-
-        return p.setIn(layer.namespace.split(':'), layerProps);
-      }, new Immutable.Map());
+        properties[layer.namespace] = layer.properties;
```

Comment:
Does this need to be
```javascript
properties[layer.namespace] = merge(properties[layer.namespace], layer.properties);
```
to match the [original behavior](https://github.com/rapid7/propsd/commit/4e925e44ab0cf09e93a2f77089b2e8a3abc468a2?diff=split#diff-327455fa29663c58312adf5955056773L129)?